frontend/public/src/entry/style.js

```diff
@@ -1 +1,7 @@
-import "../../scss/layout.scss"
+import { checkFeatureFlag } from "../lib/util"
+
+if (checkFeatureFlag("mitxonline-new-product-page")) {
```

Comment:
So I just thought (way too long) about this - and I don't think we can do this AND maintain the ability to flip flags on and off - we're back to doing a deploy to see UI changes and that's no so good. I think we might have to v2 or otherwise newly name all new scss files as well as variables.

Reply:
Not sure - it works for me in dev. I'd sorta hope(d) it'd notice the import calls and load in both sets of stylesheets but maybe not.. I can test with a local production build, though.

Reply:
So, I did some testing with a production-style build locally (basically setting environment = prod in the `.env` everywhere that it made sense to) and tested it out, and it actually did seem to work. Would you be able to test as well to verify? (I have my own Posthog account so I'll just swap back over to that locally for now so changes there won't affect the stuff I'm still working on.) 

FWIW, my criteria for success were:
- I can set the feature flag up to apply when `environment = prod` for 100% of users
- I can see the call to Posthog happen in the network pane in devtools, and see that it's sending me back a 1 (there should only be a couple of places checking the feature flag in the code - once in main app init thing and once in the product detail app)
- The Watch container remains dormant - no logs emitted during testing
- The CSS and JS bundles are coming from port *8013* rather than *8012* signifying that I'm getting a static build rather than one coming out of webpack
- (Obvs) the flag works and the design changes :-) 

This all seemed to pass (with the requisite couple of reloads when the flag status changes) but if you can give it a shot too I'd appreciate it. (Or, if there's something else I should be looking for here to make sure I'm getting a production-level build.)

Reply:
OK, now what if there are 3 feature flags - course page, home page, and... IDK potato page. Course page is set to go, because we love it and it's great, so we turn that on in prod. Now, Home page - not ready yet. We don't want that on in prod. Herein lies the issue. Yes, this works from feature to feature swapping around the imports, but can we dynamically build said styles in each of those possible T T T, TFT, TFF, FFF scenarios for those 3 pages, components, or whatever level they may be at?

We weren't using webpack when I used flags last so I'm not sure where the correct answer is outside of just... versioning class names or something simple because everything else I came up with seemed needlessly complex.

Reply:
We discussed via slack and i think the next step here is:
- Pulling out the imports from the main `loader.scss` that apply to this code
- Importing those scss files specifically within the module itself. (In this case, that'd be the ProductDetailEnrollApp component, since it should get loaded on any product page.) 
- For this specifically, there's a style that I put into the `loader.scss` (for the page background) that will move to one of the module imports

This will keep the specific styling bits that need to change separate from the main CSS. If we need to flag more stuff, we can do that and keep the flags atomic by pulling the CSS imports out of the main loader and into the components they affect. This will probably temporarily make CSS loading less efficient but since it'll be temporary it shouldn't make too much difference.